rest-interface/src/main/java/org/polypheny/db/restapi/Rest.java

```diff
@@ -0,0 +1,619 @@
+            List<RexNode> filterNodes = new ArrayList<>();
+            RelNode baseNodeForFilters = relBuilder.peek();
+            RelDataType filtersRowType = baseNodeForFilters.getRowType();
+            List<RelDataTypeField> filtersRows = filtersRowType.getFieldList();
+            for ( RequestColumn column : filters.literalFilters.keySet() ) {
+                for ( Pair<SqlOperator, Object> filterOperationPair : filters.literalFilters.get( column ) ) {
+                    int columnPosition = (int) column.getLogicalIndex();
+                    RelDataTypeField typeField = filtersRows.get( columnPosition );
+                    RexNode inputRef = rexBuilder.makeInputRef( baseNodeForFilters, columnPosition );
+                    RexNode rightHandSide = rexBuilder.makeLiteral( filterOperationPair.right, typeField.getType(), true );
+                    RexNode call = rexBuilder.makeCall( filterOperationPair.left, inputRef, rightHandSide );
+                    filterNodes.add( call );
+                }
+            }
+
+            log.debug( "Finished processing filters. Session ID: {}.", req.session().id() );
+//            relBuilder = relBuilder.filter( filterNodes );
+            log.debug( "Added filters to relation. Session ID: {}.", req.session().id() );
+            return filterNodes;
+        } else {
+            log.debug( "No filters to add. Session ID: {}.", req.session().id() );
+            return null;
+        }
+    }
+
+
+    List<String> valuesColumnNames( List<List<Pair<RequestColumn, Object>>> values ) {
+        List<String> valueColumnNames = new ArrayList<>();
+        List<Pair<RequestColumn, Object>> rowsToInsert = values.get( 0 );
+        for ( Pair<RequestColumn, Object> insertValue : rowsToInsert ) {
+            valueColumnNames.add( insertValue.left.getColumn().name );
+        }
+
+        return valueColumnNames;
+    }
+
+
+    List<List<RexNode>> valuesNode( RelBuilder relBuilder, RexBuilder rexBuilder, List<List<Pair<RequestColumn, Object>>> values, List<RelDataTypeField> tableRows ) {
+        List<List<RexNode>> wrapperList = new ArrayList<>();
+        // FIXME
+        for ( List<Pair<RequestColumn, Object>> rowsToInsert : values ) {
+            List<RexNode> rexValues = new ArrayList<>();
+            for ( Pair<RequestColumn, Object> insertValue : rowsToInsert ) {
+                int columnPosition = insertValue.left.getLogicalIndex();
+                RelDataTypeField typeField = tableRows.get( columnPosition );
+                rexValues.add( rexBuilder.makeLiteral( insertValue.right, typeField.getType(), true ) );
+            }
+            wrapperList.add( rexValues );
+        }
+
+        return wrapperList;
+    }
+
+
+    List<List<RexLiteral>> valuesLiteral( RelBuilder relBuilder, RexBuilder rexBuilder, List<List<Pair<RequestColumn, Object>>> values, List<RelDataTypeField> tableRows ) {
+        // TODO JS: FINISH THIS
```

Comment:
Please either implement or further elaborate

Reply:
Leftover comment that wasn't removed.